SPECS/mstflint/mstflint.spec

```diff
@@ -30,34 +29,105 @@ This package contains firmware update tool, vpd dump and register dump tools
 for network adapters based on Mellanox Technologies chips.
 
 %prep
-%autosetup -p1
-
-find . -type f -iname '*.[ch]' -exec chmod a-x '{}' ';'
-find . -type f -iname '*.cpp' -exec chmod a-x '{}' ';'
+%setup -q
 
 %build
-./autogen.sh
-%configure --enable-fw-mgr --enable-adb-generic-tools
-%make_build
+
+MSTFLINT_VERSION_STR="%{name} %{version}-%{release}"
+%configure --enable-fw-mgr --enable-adb-generic-tools MSTFLINT_VERSION_STR="${MSTFLINT_VERSION_STR}"
+make
 
 %install
-%make_install
-# Remove the devel files that we don't ship
-rm -fr %{buildroot}%{_includedir}
-find %{buildroot} -type f -name "*.la" -delete -print
-find %{buildroot} -type f -name '*.a' -delete
+rm -fr %{buildroot}
+make DESTDIR=%{buildroot} install
+# remove unpackaged files from the buildroot
+rm -f %{buildroot}/%{_libdir}/*.la
+# create softlinks to old mtcr header and lib locations
+# link mtcr_ul to old location
+mkdir -p %{buildroot}/%{_includedir}/mtcr_ul
+ln -s %{_includedir}/mstflint/mtcr.h %{buildroot}/%{_includedir}/mtcr_ul/mtcr.h
+ln -s %{_includedir}/mstflint/mtcr_com_defs.h %{buildroot}/%{_includedir}/mtcr_ul/mtcr_com_defs.h
+# link mtcr_ul to old lib path
+ln -s %{_libdir}/mstflint/libmtcr_ul.a %{buildroot}/%{_libdir}/libmtcr_ul.a 
+
+%clean
+rm -fr %{buildroot}
 
 %files
-%license COPYING
-%doc README
-%{_bindir}/*
-%{_sysconfdir}/mstflint
-%{_libdir}/mstflint
-
+%defattr(-,root,root)
+%{_bindir}/mstmread
+%{_bindir}/mstmwrite
+%{_bindir}/mstflint
+%{_bindir}/mstregdump
+%{_bindir}/mstmtserver
+%{_bindir}/mstvpd
+%{_bindir}/mstmcra
+%{_bindir}/mstconfig
+%{_bindir}/mstfwreset
+%{_bindir}/mstcongestion
+%{_bindir}/mstprivhost
+%{_bindir}/mstfwtrace
+%{_bindir}/mstresourcedump
+%{_bindir}/mstresourceparse
+ %{_bindir}/mstfwmanager
```

Comment:
There are blank spaces on this line, and at the start of a few other lines. Can you please remove them?